test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +178,298 @@ int suiteTearDown( int numFailures )
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 1, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_TRUE( returnFlags.addAcknowledgment );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Unsubscribe() works as intended.
+ */
+void test_MQTTAgentCommand_Unsubscribe( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentSubscribeArgs_t subscribeArgs = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_GetPacketId_IgnoreAndReturn( 1 );
+    MQTT_Unsubscribe_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Unsubscribe( &mqttAgentContext, &subscribeArgs, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 1, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_TRUE( returnFlags.addAcknowledgment );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Disconnect() works as intended.
+ */
+void test_MQTTAgentCommand_Disconnect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Disconnect_ExpectAndReturn( &mqttAgentContext, MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Disconnect( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_TRUE( returnFlags.endLoop );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Ping() works as intended.
+ */
+void test_MQTTAgentCommand_Ping( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Ping_ExpectAndReturn( &mqttAgentContext, MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Ping( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Connect() success case.
+ */
+void test_MQTTAgentCommand_Connect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentConnectArgs_t connectInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Connect_IgnoreAndReturn( MQTTSuccess );
+    MQTTAgent_ResumeSession_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Connect( &mqttAgentContext, &connectInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
```

Comment:
Test all the return flags when they're relevant, not just the ones that are set. Here, we should also make sure that `addAcknowledgment` and `endLoop` are false.